model/src/main/kotlin/ScanResult.kt

```diff
@@ -49,4 +55,53 @@ data class ScanResult(
     @JsonAlias("rawResult")
     @JsonInclude(JsonInclude.Include.NON_NULL)
     val rawResult: JsonNode? = null
-)
+) {
+    /**
+     * Filter all detected licenses and copyrights from the [summary] which are below [path], and set the [path] for
+     * [provenance]. Findings in files which are listed in [LICENSE_FILE_NAMES] are also kept.
+     */
+    fun filterPath(path: String): ScanResult {
+        val matchersForLicenseFiles = LICENSE_FILE_NAMES.map {
+            FileSystems.getDefault().getPathMatcher("glob:$it")
+        }
+
+        val pathToMatch = Paths.get(path)
+
+        fun SortedSet<TextLocation>.filterPath() =
+            filterTo(sortedSetOf()) { location ->
+                location.path.startsWith("$path/") || matchersForLicenseFiles.any { matcher ->
+                    matcher.matches(pathToMatch)
+                }
+            }
+
+        val newProvenance = provenance.copy(
+            vcsInfo = provenance.vcsInfo?.copy(path = path),
+            originalVcsInfo = provenance.originalVcsInfo?.copy(path = path)
+        )
+
+        val findings = summary.licenseFindings.mapNotNull { finding ->
+            val locations = finding.locations.filterPath()
+
+            if (locations.isNotEmpty()) {
+                val copyrights = finding.copyrights.mapNotNull { copyright ->
+                    val copyrightLocations = copyright.locations.filterPath()
+
+                    if (copyrightLocations.isNotEmpty()) {
+                        CopyrightFinding(copyright.statement, copyrightLocations)
+                    } else {
+                        null
+                    }
+                }.toSortedSet()
+
+                LicenseFinding(finding.license, locations, copyrights)
+            } else {
+                null
+            }
+        }.toSortedSet()
+
+        val fileCount = findings.flatMap { finding -> finding.locations.map { it.path } }.size
```

Comment:
The problem is that at this point we do not know how many files are in the project directory. To me it made more sense to use the number of files with license findings than the number of files in the whole repository. Given that we do not use the file count anywhere I think we should either remove it or add a list with all files (not only those with license findings) to the scan result, so that we can calculate it accurately.

Reply:
Removing sounds reasonable as it seems as YAGNI.